### DISCUSSION TITLE
Iterate over all the boundary conditions in the case

### DISCUSSION THREAD

**Discussion by rachelzhang333** [ansys/pyfluent#1636] [Q&A] [answered]
I'd like to get all boundary conditions and report definitions in the case. How do I achieve it?

**Comment by h-krishnan** [comment, marked as answer]
```
for bname, bobj in solver.setup.boundary_conditions.items():
    print (bname, bobj.get_state())

for rname, robj in solver.solution.report_definitions.items():
   print (rname, robj.get_state())
```

**Reply by rachelzhang333** [reply]
I tried to use boundary_conditions.items. It returns Attribute Error. It seems there is no items.
PS :My Fluent version is 22.2.0
![1684996846094](https://github.com/ansys/pyfluent/assets/134527835/609410c1-e0b9-4cd7-b2dd-1774d60bf0c0)

**Comment by dnwillia-work** [comment]
Yes, important to say what version of the library and Fluent you are using :-).  I think it should work with 23.1 and the latest library release.

**Reply by h-krishnan** [reply]
In 22.2, this should work:
``` python
>>> def get_all_boundary_objects(setup):
...    for btype in setup.boundary_conditions.get_active_child_names():
...       for bname, bobj in getattr(setup.boundary_conditions, btype).items():
...          yield bname, bobj
...
>>> for bname, bobj in get_all_boundary_objects(solver.setup):
...    print (bname, bobj.get_state())
...

**Reply by rachelzhang333** [reply]
It works. Thanks a lot
